### PR TITLE
Update/cmake-build-refactoring

### PR DESCRIPTION
**Pull request by EricJIsaac**
**Pull request recommendations:**

- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-##], adds tiff file format support_
- [x] Provide description and context of changes.
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

This PR: 
-  Updates the aws-cpp-sdk
-  Reorganizes the build config so that fewer of the compilation targets interfere with each other

Resolves:
https://github.com/allen-cell-animated/simularium-engine/issues/15
https://github.com/allen-cell-animated/simularium-engine/issues/16

More details:
GCC 11 added warning for several common memory/compile errors. The way the simularium-engine build was setup, warnings are treated as errors. More generally, the compilation environment reflected the strictest settings from any dependency.

To address this, the compilation environment for the dependencies (simulation engines) and the server have been separated further. There is still overlap, but it is a step in the right direction. Further, the aws-sdk-cpp was updated to receive changes that allowed us more control over the compilation environment  for the sdk.

Autorun tests, in the sdk, were disabled, since there are known issues with some of the newer tests in untested environments (e.g. Docker images)
https://github.com/aws/aws-sdk-cpp/issues/1601

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,14 +12,14 @@ RUN mkdir /agentsim-dev && \
 	cmake \
 	curl \
 	git \
-    openssl \
+  openssl \
```
Comment (toloudis) [inline Dockerfile:15]: indenting is off

File: Dockerfile [hunk 2]
```diff
@@ -12,14 +12,14 @@ RUN mkdir /agentsim-dev && \
...
 	libblas-dev \
 	libhdf5-dev \
 	liblapack-dev \
 	python-dev \
 	libssl-dev libcurl4-openssl-dev \
 	libblosc1 \
-    libglew-dev mesa-common-dev freeglut3-dev
+  libglew-dev mesa-common-dev freeglut3-dev
```
Comment (toloudis) [inline Dockerfile:22]: indenting is off

File: Dockerfile [hunk 3]
```diff
@@ -30,9 +30,9 @@ RUN git submodule update --init --recursive
 
 # build agentsim project
 RUN cd ../build && \
-	cmake ../agentsim -DBUILD_ONLY="s3;awstransfer;transfer" -DCMAKE_BUILD_TYPE=Release && \
+	cmake ../agentsim -DBUILD_ONLY="s3;awstransfer;transfer" -DCMAKE_BUILD_TYPE=Release -DAUTORUN_UNIT_TESTS=OFF && \
 	make && \
-    openssl dhparam -out /dh.pem 2048 && \
+  openssl dhparam -out /dh.pem 2048 && \
```
Comment (toloudis) [inline Dockerfile:35]: indenting is off

File: src/test/CMakeLists.txt [hunk 4]
```diff
@@ -17,18 +16,14 @@ set(TEST_INCLUDES
...
-    "${DEPENDENCY_DIRECTORY}/aws-sdk-cpp/aws-cpp-sdk-s3/include"
     "${DEPENDENCY_DIRECTORY}/gtest/googletest/include"
     "${OPENSSL_INCLUDES}"
     "/usr/local/opt/openssl/include"
 )
 
 set(TEST_LIBS
     "${TARGET}"
-    Threads::Threads
+    #Threads::Threads
```
Comment (toloudis) [inline src/test/CMakeLists.txt:26]: can remove?

**Review by toloudis — APPROVED** [review]
Looks good, looks like it is moving in the right direction.  One question:  is the aws sdk being staticly linked or dynamically linked?

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -1,2 +1,58 @@
...
+"${EXTERNAL_DIRECTORY}/asio"
+"/usr/local/opt/openssl/include"
+)
+
+target_link_libraries(${SERVER_PROGRAM}
+    "${TARGET}"
+    Threads::Threads
+)
+
+set(CLIENT_PROGRAM_SOURCES
```
Comment (toloudis) [inline src/CMakeLists.txt:35]: what does CLIENT_PROGRAM do?

**Comment by EricJIsaac** [comment]
> Looks good, looks like it is moving in the right direction. One question: is the aws sdk being staticly linked or dynamically linked?

The aws sdk is dynamically linked. Why?

File: src/CMakeLists.txt (hunk 5 above)
Comment (EricJIsaac) [reply src/CMakeLists.txt:35]: A program for testing, it receives client messages and can issue commands through a CLI.

**Comment by toloudis** [comment]
> > Looks good, looks like it is moving in the right direction. One question: is the aws sdk being staticly linked or dynamically linked?
> 
> The aws sdk is dynamically linked. Why?

Looser coupling if it's dynamically linked.  The build doesn't necessarily need to be as entangled with the other components and could be practically independent (depending on allocation patterns and data passing across the interface, but the chances are good).

**Pull request merged**